src/Chapter3.hs

```diff
@@ -907,9 +999,25 @@ Implement instances of "Append" for the following types:
   ✧ *(Challenge): "Maybe" where append is appending of values inside "Just" constructors
 
 -}
+newtype Gold = Gold Int
+
 class Append a where
     append :: a -> a -> a
 
+instance Append Gold where
+    append :: Gold -> Gold -> Gold
+    append (Gold a) (Gold b) = Gold (a + b)
+
+instance Append [a] where
+    append :: [a] -> [a] -> [a]
+    append xs ys = xs <> ys
+
+instance (Semigroup a) => Append (Maybe a) where
+    append :: Maybe a -> Maybe a -> Maybe a
+    append Nothing (Just a) = Just a
+    append (Just a) Nothing = Just a
+    append (Just a) (Just b) = Just (a <> b)
+    append _ _ = Nothing
```

Comment:
I think this case could be avoided if you generilise first two 🙂 
You can notice that in case of one `Nothing` the result is always the other argument 👌🏼